### DISCUSSION TITLE
Overwrite cluster.local suffix

### DISCUSSION THREAD

**Discussion by m4rc3l-h3** [authelia/chartrepo#221] [Q&A] [answered]
Hi everyone,

I am trying to deploy the Helm chart on a cluster which has been configured to use `xx.cluster.home` instead of `xx.cluster.local`. That causes some issues with the creation of traefik `Middleware` resouces, in particular the `forwardAuth.address`. I could track it down to line 574 in [_helpers.tpl](https://github.com/authelia/chartrepo/blob/5eac62d65f3a8d3a154c7f0f3fba8c70ec117d45/charts/authelia/templates/_helpers.tpl#L574).

I am not that familiar with Helm and was wondering if and where to set `.Cluster` to a custom value. Is that possible within `values.yaml`?

Any guidance would be much appreciated. Thank you for your support!
If that is not the right place to ask, please let me know. Thanks.

**Comment by james-d-elliott** [comment]
Can't say I've ever seen this before. I don't recall adding this. I'll take a look at it for 0.9.0 which will bundle 4.38.0. I could use some additional testers to be fair as it's a fairly major rework.

**Reply by m4rc3l-h3** [reply]
Thank you for your swift response. Sure I can and will test it.
As far as I can see in the current version of files in the repo, it is the only place where it is used. In addition, it is also the only place where it causes issues in my current deployment as other components (e.g., mysql, redis) allow to specify the hostname directly. However, I will have a closer look and do some more testing and add findings to this thread, if that is okay?

Maybe an idea would be to add `clusterDomain` to `values.yaml` and use that in the `helper.tpl`? This is, for example, done by some bitnami charts I am using: [mongodb](https://github.com/bitnami/charts/blob/91b7e4bdf9c6dda6ec1dc6a951064aa7dc2c8496/bitnami/mongodb/values.yaml#L50), [postgresql](https://github.com/bitnami/charts/blob/91b7e4bdf9c6dda6ec1dc6a951064aa7dc2c8496/bitnami/postgresql/values.yaml#L68)?

I could give it a try, if that is okay and you could provide some guidance.

Thank you again for your feedback, much appreciated.

**Reply by james-d-elliott** [reply, marked as answer]
Available in https://github.com/authelia/chartrepo/pull/215/commits/a802da81af36ea78283465a6cd13b8f493a626f1, see https://github.com/authelia/chartrepo/pull/215

**Reply by m4rc3l-h3** [reply]
Hi @james-d-elliott 
Sorry for the late reply and thank you again for looking into it/adding the option. I am currently testing 0.9.0-beta5 and the `kubeDNSDomainOverride` worked for the use case described above. Still working on the complete setup, but thank you very much for your support.
